src/state/slices/opportunitiesSlice/thunks.ts

```diff
@@ -0,0 +1,105 @@
+            },
+            // Any previous query without portfolio loaded will be rejected, the first successful one will be cached
+            { forceRefetch: false },
+          ),
+        ),
+    ),
+  )
+}
+
+export const fetchAllStakingOpportunitiesMetadata = async () => {
+  const { getOpportunityMetadata } = opportunitiesApi.endpoints
+
+  await Promise.all(
+    foxEthStakingIds.map(
+      async opportunityId =>
+        await store.dispatch(
+          getOpportunityMetadata.initiate(
+            {
+              opportunityId: opportunityId as LpId,
+              opportunityType: DefiType.Staking,
+              defiType: DefiType.Staking,
+            },
+            // Any previous query without portfolio loaded will be rejected, the first successful one will be cached
+            { forceRefetch: false },
+          ),
+        ),
+    ),
+  )
+}
+
+export const fetchAllOpportunitiesMetadata = async () => {
+  // Don't Promise.all() me - parallel execution would be better, but the market data of the LP tokens gets populated when fetching LP opportunities
+  // Without it, we won't have all we need to populate the staking one - which is relying on the market data of the staked LP token for EVM chains LP token farming
+  await fetchAllLpOpportunitiesMetadata()
+  await fetchAllStakingOpportunitiesMetadata()
+}
+
+export const fetchAllStakingOpportunitiesUserData = async (accountId: AccountId) => {
+  const { getOpportunityUserData } = opportunitiesApi.endpoints
+
+  await Promise.all(
+    foxEthStakingIds.map(
+      async opportunityId =>
+        await store.dispatch(
+          getOpportunityUserData.initiate(
+            {
+              accountId,
+              opportunityId: opportunityId as LpId,
+              opportunityType: DefiType.Staking,
+              defiType: DefiType.Staking,
+            },
+            // Any previous query without portfolio loaded will be rejected, the first successful one will be cached
+            { forceRefetch: false },
+          ),
+        ),
+    ),
+  )
+}
+
+export const fetchAllLpOpportunitiesUserdata = async (accountId: AccountId) => {
+  const { getOpportunityUserData } = opportunitiesApi.endpoints
+
+  await Promise.all(
+    foxEthLpAssetIds.map(
+      async opportunityId =>
+        await store.dispatch(
+          getOpportunityUserData.initiate(
+            {
+              accountId,
+              opportunityId: opportunityId as LpId,
+              opportunityType: DefiType.LiquidityPool,
+              defiType: DefiType.LiquidityPool,
+            },
+            // Any previous query without portfolio loaded will be rejected, the first succesful one will be cached
+            { forceRefetch: false },
+          ),
+        ),
+    ),
+  )
+}
+
+export const fetchAllOpportunitiesUserData = async (accountId: AccountId) => {
+  await fetchAllLpOpportunitiesUserdata(accountId)
+  await fetchAllStakingOpportunitiesUserData(accountId)
```

Comment:
We should be able to `Promise.all()` these and not block execution on LP opportunities resolution

Reply:
in general i think we want to trend towards `.allSettled` with logging and error handling rather than `.all`

Reply:
Disregard me, we actually need to block execution on `fetchAllLpOpportunitiesUserdata` tick